mkosi

```diff
@@ -882,6 +888,78 @@ gpgkey={gpg_key}
         ["basesystem-minimal"],
         ["kernel-server-latest", "binutils"])
 
+@complete_step('Installing CentOS')
+def install_centos(args, workspace, run_build_script):
+
+    disable_kernel_install(args, workspace)
+
+    gpg_key = "/etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-%s" % args.release
+    if os.path.exists(gpg_key):
+        gpg_key = "file://%s" % gpg_key
+    else:
+        gpg_key = "https://www.centos.org/keys/RPM-GPG-KEY-CentOS-%s" % args.release
+
+    if args.mirror:
+        release_url = "baseurl={args.mirror}/centos/{args.release}/os/x86_64".format(args=args)
+        updates_url = "baseurl={args.mirror}/cenots/{args.release}/updates/x86_64/".format(args=args)
+    else:
+        release_url = "mirrorlist=http://mirrorlist.centos.org/?release={args.release}&arch=x86_64&repo=os".format(args=args)
+        updates_url = "mirrorlist=http://mirrorlist.centos.org/?release={args.release}&arch=x86_64&repo=updates".format(args=args)
+
+    with open(os.path.join(workspace, "yum.conf"), "w") as f:
+        f.write("""\
+[main]
+gpgcheck=1
+
+[base]
+name=CentOS-{args.release} - Base
+{release_url}
+gpgkey={gpg_key}
+
+[updates]
+name=CentOS-{args.release} - Updates
+{updates_url}
+gpgkey={gpg_key}
+""".format(args=args,
+           gpg_key=gpg_key,
+           release_url=release_url,
+           updates_url=updates_url))
+
+    repos = ["--enablerepo=" + repo for repo in (args.repositories if args.repositories else ["base", "updates"])]
+
+    root = os.path.join(workspace, "root")
+    cmdline = ["yum",
+               "-y",
+               "--config=" + os.path.join(workspace, "yum.conf"),
+               "--best",
+               "--allowerasing",
+               "--releasever=" + args.release,
+               "--installroot=" + root,
+               "--disablerepo=*",
+               *repos,
+               "--setopt=keepcache=1",
+               "--setopt=install_weak_deps=0"]
+
```

Comment:
It's not quite the same. I know that `install_weak_deps` isn't the correct option here. I think for Yum, it's `requires_policy`, but I don't recall what the params are.